### PR TITLE
Extract EventTracker logic to the appsec module

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Extracts `datadog.trace.api.EventTracker` logic and moves it into the appsec module, the actual implementation will be injected at runtime by the `com.datadog.appsec.AppSecSystem#start` method during initialization. Since these classes are added to the boostrap classloader by the tracer, customers will always be using the latest version provided by the tracer.

# Motivation
Moving the logic into the appsec module makes it easier to modify its behavior and make use of other appsec tools like the WAF which is needed for attacker fingerprinting.

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/user/AppSecEventTrackerImpl.java
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/appsec/src/main/java/com/datadog/appsec/user/AppSecEventTrackerImpl.java:1]: Nothing new here, just a mix between the old `EventTracker` and `AppSecUserEventDecorator`

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/user/AppSecEventTrackerSpecification.groovy
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/user/AppSecEventTrackerSpecification.groovy:1]: Nothing new here just a mix between `EventTrackerTest` and `AppSecUserEventDecoratorTest`

**Review by jandro996 — APPROVED** [review]
LGTM! It wuld be nice if could improve the PR description :)

**Review by smola — APPROVED** [review]
Looks good, but please, update the PR description. Especially explaining why this works with old SDK dependencies because of classloaders, etc.

**Pull request merged**